File: LeetCode/C/24-linked-list-两两交换链表中的节点.cpp
/*
 * @lc app=leetcode.cn id=24 lang=cpp
 *
 * [24] 两两交换链表中的节点
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    ListNode *swapPairs(ListNode *head) {
        if (head == nullptr || head->next == nullptr)
            return head;
        ListNode *a = head, *b = head->next;
        ListNode *t = b->next;
        b->next = a;
        a->next = swapPairs(t);
        return b;
    }
};
// @lc code=end

File: LeetCode/Go/24-linked-list-两两交换链表中的节点.go
/*
 * @lc app=leetcode.cn id=24 lang=golang
 *
 * [24] 两两交换链表中的节点
 */
package leetcode

// @lc code=start
func swapPairs(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	pre, cur := head, head.Next
	t := cur.Next
	cur.Next = pre
	pre.Next = swapPairs(t)
	return cur
}

// @lc code=end
